TMS.Net07.Homework3.CurrencyAndCalculator/TMS.Net07.Homework3.High.TextCalculator/TMS.Net07.Homework3.High.TextCalculator/Program.cs

```diff
@@ -0,0 +1,229 @@
+                    break;
+                case "sqrt":
+                    if (argAfterMathOperDouble >= 0)
+                    {
+                        result = Math.Sqrt(argAfterMathOperDouble);
+                        Console.WriteLine("Result: {0}", result);
+                    }
+                    else
+                    {
+                        Console.WriteLine("No! No! No! It is wrong. Argument should be >= 0");
+                    }
+                    break;
+                case "cos":
+                    if ((argAfterMathOperDouble >= 0) && (argAfterMathOperDouble <= 360))
+                    {
+                        double radian = (argAfterMathOperDouble * Math.PI) / 180;
+                        result = Math.Cos(radian);
+                        Console.WriteLine("Result: {0}", result);
+                    }
+                    else
+                    {
+                        Console.WriteLine("No! No! No! It is wrong. Argument should be >= 0 and <= 360");
+                    }
+                    break;
+                case "sin":
+                    if ((argAfterMathOperDouble >= 0) && (argAfterMathOperDouble <= 360))
+                    {
+                        double radian = (argAfterMathOperDouble * Math.PI) / 180;
+                        result = Math.Sin(radian);
+                        Console.WriteLine("Result: {0}", result);
+                    }
+                    else
+                    {
+                        Console.WriteLine("No! No! No! It is wrong. Argument should be >= 0 and <= 360");
+                    }
+                    break;
+                case "pow":
+                    result = Math.Pow(argBeforeMathOperDouble, argAfterMathOperDouble);
+                    Console.WriteLine("Result: {0}", result);
+                    break;
+                case "+":
+                    result = argBeforeMathOperDouble + argAfterMathOperDouble;
+                    Console.WriteLine("Result: {0}", result);
+                    break;
+                case "plus":
+                    result = argBeforeMathOperDouble + argAfterMathOperDouble;
+                    Console.WriteLine("Result: {0}", result);
+                    break;
+                case "-":
+                    result = argBeforeMathOperDouble - argAfterMathOperDouble;
+                    Console.WriteLine("Result: {0}", result);
+                    break;
+                case "minus":
+                    result = argBeforeMathOperDouble - argAfterMathOperDouble;
+                    Console.WriteLine("Result: {0}", result);
+                    break;
+                case "*":
+                    result = argBeforeMathOperDouble * argAfterMathOperDouble;
+                    Console.WriteLine("Result: {0}", result);
+                    break;
+                case "multiply":
+                    result = argBeforeMathOperDouble * argAfterMathOperDouble;
```

Comment:
To avoid code duplication we could do:
```
case "*":
case "multiply":
case "x":
case "X":
       result = argBeforeMathOperDouble * argAfterMathOperDouble;
       Console.WriteLine("Result: {0}", result);
       break;
```

Reply:
Ok, I all fixed